### PR TITLE
feat(v1.x.x): async pruning of orphan nodes

### PR DESCRIPTION
**Pull request by czarcas7ic**
This feature prevents the multiple hour long waiting period when chains upgrade from previous versions of IAVL to IAVL v1. The time comes from pruning orphan nodes. This synchronously prunes them. This branch has been tested successfully against osmosis mainnet.

### CODE REVIEW

**Comment by cool-develope** [comment]
overall, looks good to me! could you please add the changelog?

**Comment by czarcas7ic** [comment]
Will add now. @cool-develope would you consider making a minor v1 release that includes this? If not we can just use our fork but would be nice to just use a tag from iavl proper.

**Review by ValarDragon — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Improvements
+
+- [#876](https://github.com/cosmos/iavl/pull/876) Make pruning of orphan nodes synchronous.
```
Comment (cool-develope) [inline CHANGELOG.md:7]:
```suggestion
- [#876](https://github.com/cosmos/iavl/pull/876) Make pruning of legacy orphan nodes asynchronous.
```

**Comment by czarcas7ic** [comment]
@cool-develope would we be able to get this in a v1 release? I am also refraining from merging since I am not aware of the rules/requirements for this repo.

**Comment by cool-develope** [comment]
@tac0turtle ☝️

**Comment by tqin7** [comment]
Thanks for working on this! @cool-develope are you looking to create a minor release off this and if so, may I ask when we should expect it?

**Comment by tac0turtle** [comment]
we can do a release this week for it

**Comment by BrendanChou** [comment]
Nit: the PR description should say "asynchronously" ?

**Pull request merged**